gatling-http/src/main/scala/io/gatling/http/protocol/HttpProtocol.scala

```diff
@@ -181,7 +182,8 @@ final case class HttpProtocolResponsePart(
 
 final case class HttpProtocolWsPart(
     wsBaseUrls: List[String],
-    maxReconnects: Int
+    maxReconnects: Int,
+    autoReplyTextFrames: List[PartialFunction[String, String]]
```

Comment:
So we can react to multiple different messages. Ping/pong is one kind of a message, but I imagine there could be others like that, even app specific, so wanted to support that as well. So something like this works:
```
    .wsAutoReplyTextFrame({ case "ping" => "pong" })
    .wsAutoReplyTextFrame({ case "1" => "2" })
    .wsAutoReplySocketIo4
```

Reply:
Nah, writing this this way would be a waste of partial functions.

```
.wsAutoReplyTextFrame(
{
  case "ping" => "pong" 
  case "1" => "2"
  case "2" => "3"
})
```

Reply:
I see, makes more sense :sweat_smile: 
I'll update.

Reply:
thanks :)

Reply:
The documentation says:

> It is the responsibility of the caller to call isDefinedAt before calling apply, because if isDefinedAt is false, it is not guaranteed apply will throw an exception to indicate an error condition. If an exception is not thrown, evaluation may result in an arbitrary value.

https://www.scala-lang.org/api/2.13.3/scala/PartialFunction.html

So I'm not sure if try/catch will work here.

Reply:
We really don't want to pay the overhead of creating Exceptions.
We'll use isDefinedAt before call it.

if (f.isDefinedAt(text)) {
  // send reply
} else {
  // log unmatched message
}